.github/workflows/ci.yml

```diff
@@ -0,0 +1,31 @@
+---
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+      - master
```

Comment:
@jamtur01 I added `main` assuming we might want to change the default branch in this project too.

Reply:
+1

Reply:
Thanks, I amended this commit and rebased on top of main. This is ready for review!